Source/WebCore/platform/gstreamer/WebKitFliteSourceGStreamer.cpp

```diff
@@ -0,0 +1,263 @@
+    return TRUE;
+}
+
+static GstFlowReturn webkitFliteSrcCreate(GstBaseSrc* baseSource, guint64 offset, guint length, GstBuffer** buffer)
+{
+    UNUSED_PARAM(offset);
+    UNUSED_PARAM(length);
+
+    WebKitFliteSrc* src = WEBKIT_FLITE_SRC(baseSource);
+    WebKitFliteSrcPrivate* priv = src->priv;
+
+    gsize bytes = DEFAULT_SAMPLES_PER_BUFFER * sizeof(gint16) * priv->info.channels;
+    DataMutexLocker members { priv->dataMutex };
+    if (!members->didLoadUtterance) {
+        members->didLoadUtterance = true;
+
+        cst_wave* wave = flite_text_to_wave(priv->text.utf8().data(), priv->voice);
+        cst_wave_resample(wave, priv->info.rate);
+
+        GST_DEBUG("type %s, sample_rate %d, num_samples %d, num_channels %d",
+            wave->type, wave->sample_rate, wave->num_samples, wave->num_channels);
+
+        gsize bufferSize = priv->info.channels * sizeof(gint16) * wave->num_samples;
+        GRefPtr<GstBuffer> buf = adoptGRef(gst_buffer_new_allocate(nullptr, bufferSize, nullptr));
+        GstMappedBuffer map(buf, GST_MAP_WRITE);
+        gint16* data = reinterpret_cast<gint16*>(map.data());
+        memset(data, 0, bufferSize);
+        for (int i = 0; i < wave->num_samples; i++)
+            data[i * priv->info.channels] = wave->samples[i];
+
+        gst_adapter_push(members->adapter.get(), gst_buffer_ref(buf.get()));
+    }
+
+    size_t queueSize = gst_adapter_available(members->adapter.get());
+    if (members->didLoadUtterance && !queueSize)
+        return GST_FLOW_EOS;
+
+    *buffer = gst_adapter_take_buffer(members->adapter.get(), std::min(queueSize, bytes));
+    return GST_FLOW_OK;
+}
+
+static gboolean webkitFliteSrcSetCaps(GstBaseSrc* baseSource, GstCaps* caps)
+{
+    WebKitFliteSrc* src = WEBKIT_FLITE_SRC(baseSource);
+    WebKitFliteSrcPrivate* priv = src->priv;
+
+    gst_audio_info_init(&priv->info);
+    if (!gst_audio_info_from_caps(&priv->info, caps)) {
+        GST_ERROR_OBJECT(src, "Invalid caps");
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+static void webkit_flite_src_class_init(WebKitFliteSrcClass* klass)
+{
+    GObjectClass* objectClass = G_OBJECT_CLASS(klass);
+    objectClass->constructed = webkitFliteSrcConstructed;
+
+    GstElementClass* elementClass = GST_ELEMENT_CLASS(klass);
+    elementClass->change_state = webkitFliteSrcChangeState;
+
+    gst_element_class_add_static_pad_template(elementClass, &srcTemplate);
+
+    gst_element_class_set_static_metadata(elementClass,
+        "WebKit WebSpeech GstFlite source element", "Source",
+        "Handles WebSpeech data from WebCore",
+        "ChangSeok Oh <changseok@webkit.org>");
+
+    GstBaseSrcClass* baseSrcClass = GST_BASE_SRC_CLASS(klass);
+    baseSrcClass->start = GST_DEBUG_FUNCPTR(webkitFliteSrcStart);
+    baseSrcClass->stop = GST_DEBUG_FUNCPTR(webkitFliteSrcStop);
+    baseSrcClass->create = GST_DEBUG_FUNCPTR(webkitFliteSrcCreate);
+    baseSrcClass->set_caps = GST_DEBUG_FUNCPTR(webkitFliteSrcSetCaps);
+
+    flite_init();
```

Comment:
Maybe move this to a once init in the `constructed` func?

Reply:
O.K. Moved into ensureFliteVoicesInitialized()